src/FSharp.Core/Random.fsi

```diff
@@ -0,0 +1,7 @@
+namespace Microsoft.FSharp.Core
+
+open System
+
+[<AbstractClass; Sealed>]
+type internal ThreadSafeRandom =
```

Comment:
I'd say... **thread-safe** is 1) an implementation detail 2) a must if we are talking about the core language features.

So I'd keep this just `Random`. Keeping `ThreadSafe` in the API surface suggests there is another, non-thread-safe, random :)

Reply:
I've tried to rename it and realized that it might be not a good idea: whenever we meet `Random` in the codebase it would be unclear if it is Microsoft.Fsharp.Core Random or System.Random.
So, if you don't like `ThreadSafeRandom`, we can go with something else. Maybe `InternalRandom` or `RandomWrapper` or `RandomAccessor`.

Reply:
We are speaking about an `internal` type here, this does not leak into any API surface.
I think the usage is appropriate, since this is what it is - a thread safe accessor to values of `Random`.

Reply:
Well it's still an API surface, just within the assembly :shrug:

I like RandomWrapper more, it seems less misleading to me.

Reply:
The BCL uses the name `ThreadSafeRandom` for the equivalent of this, for what it's worth: https://github.com/dotnet/runtime/blob/fb1fe7f3481fc88fa477268c18d60d824b83c314/src/libraries/System.Private.CoreLib/src/System/Random.cs#L360-L373

Reply:
Alright, consistency above all, don't mind keeping the name anymore :D